sdk/msp-client/src/modules/buckets.ts

```diff
@@ -1,10 +1,41 @@
 import type { MspClientContext } from "../context.js";
+import type { Bucket, FileListResponse, GetFilesOptions } from "../types.js";
 import { ModuleBase } from "../base.js";
 
 export class BucketsModule extends ModuleBase {
   constructor(ctx: MspClientContext) {
     super(ctx);
   }
+
+  /** List all buckets for the current authenticated user */
+  list(options?: { signal?: AbortSignal }): Promise<Bucket[]> {
+    const headers = this.withAuth();
+    return this.ctx.http.get<Bucket[]>("/buckets", {
+      ...(headers ? { headers } : {}),
+      ...(options?.signal ? { signal: options.signal } : {})
+    });
+  }
+
+  /** Get a specific bucket's metadata by its bucket ID */
+  get(bucketId: string, options?: { signal?: AbortSignal }): Promise<Bucket> {
+    const headers = this.withAuth();
+    const path = `/buckets/${encodeURIComponent(bucketId)}`;
+    return this.ctx.http.get<Bucket>(path, {
+      ...(headers ? { headers } : {}),
+      ...(options?.signal ? { signal: options.signal } : {})
+    });
+  }
+
+  /** List files/folders under a path for a bucket (root if no path) */
+  getFiles(bucketId: string, options?: GetFilesOptions): Promise<FileListResponse> {
+    const headers = this.withAuth();
+    const path = `/buckets/${encodeURIComponent(bucketId)}/files`;
+    return this.ctx.http.get<FileListResponse>(path, {
+      ...(headers ? { headers } : {}),
+      ...(options?.signal ? { signal: options.signal } : {}),
+      ...(options?.path ? { query: { path: options.path.replace(/^\/+/, "") } } : {})
```

Comment:
Could you add a comment explaining what this replace does, in words? I'm not that good at regex, but to me it seems like it removed all `/` instances at the start of the string, right?
So `/foo/bar` becomes `foo/bar`?